private_include/utils_private_cxx.hpp

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+namespace
+{
+template<typename DriverType, typename CxxType>
+DriverType cxx_to_driver_type(const CxxType& cxx_type)
```

Comment:
I'd suggest a shorter, more generic name, e.g. `cxx_to_type` or `original_type`. Shorter to save typing, hence encouraging more users to use it; more generic because you can use that for strong value type, not just driver types.